stacks-signer/src/runloop.rs

```diff
@@ -235,12 +236,35 @@ impl RunLoop {
                     }
                 }
             }
-            let new_signer = Signer::from(new_signer_config);
+            let mut new_signer = Signer::from(new_signer_config);
+
+            let dkg_id = retry_with_exponential_backoff(|| {
+                self.stacks_client
+                    .get_last_round(reward_cycle)
+                    .map_err(backoff::Error::transient)
+            })?
+            .unwrap_or(0);
+
+            let approved_aggregate_key = retry_with_exponential_backoff(|| {
+                self.stacks_client
+                    .get_approved_aggregate_key(reward_cycle)
+                    .map_err(backoff::Error::transient)
+            })?;
+
+            new_signer
+                .state_machine
+                .reset(dkg_id, &mut storage::crypto_rng());
+            new_signer.approved_aggregate_public_key = approved_aggregate_key;
+            new_signer
+                .load_saved_state()
```

Comment:
I think you'll want to check that `approved_aggregate_key` is consistent with the private state you load here?

Reply:
I'll make the `load_saved_state()` function take the aggregate key as input instead so the caller don't have to worry about this.

Reply:
Removed this code, and changed to use only the `load_saved_state_for_aggregate_key` method in the only place we load state now.